### DISCUSSION TITLE
reflect schema to model

### DISCUSSION THREAD

**Discussion by techolas23** [piccolo-orm/piccolo#190] [Q&A]
does piccolo require a model first or can i reflect a model based on the metadata? if not, would it be of interest?

**Comment by dantownsend** [comment]
Currently the models need to defined by hand.

There's a pull request in progress which adds basic support for auto generating models from an existing database:

https://github.com/piccolo-orm/piccolo/pull/157

I hope to merge it in today. However, there's still plenty of scope for it to be improved if you're interested in this area.
